src/ASM/ASMs2D.C

```diff
@@ -2651,10 +2653,12 @@ void ASMs2D::generateThreadGroups (size_t strip1, size_t strip2, bool silence)
 #endif
       // Verify that the nodes on this thread are not present on the others
       for (k = 0; k < j; k++)
-        for (nit = nodes[j].begin(); nit != nodes[j].end(); ++nit)
-          if (nodes[k].find(*nit) != nodes[k].end())
+        for (nit = nodes[j].begin(); nit != nodes[j].end(); ++nit) {
+          if ((!this->isLMn(*nit+1) || this->getLMType(*nit+1) != 'G' || !ignoreGlobalLM) &&
+               nodes[k].find(*nit) != nodes[k].end())
```

Comment:
ja, i de aller fleste tilfellene vil man jo det. med mindre koden er skrevet for å takle det kan du ikke kjøre multithreading med en global multiplikator. denne fins på *ALLE* element, så diagonal-elementet og (mer relevant) vektor-entryen vil da alltid være racy. bracene er en artifakt fra en logging jeg hadde under debugging, skal fjerne. og var usikker på om getLMType sjekket for multiplikator og gjorde det sikre, skal fjerne.